src/Common/MemoryTracker.cpp

```diff
@@ -68,7 +68,7 @@ MemoryTracker::MemoryTracker(MemoryTracker * parent_, VariableContext level_) :
 
 MemoryTracker::~MemoryTracker()
 {
-    if ((level == VariableContext::Process || level == VariableContext::User) && peak)
+    if ((level == VariableContext::Process || level == VariableContext::User) && peak && log_peak_memory_usage_in_destructor)
```

Comment:
We log peak memory usage only every 1 GB, that's why it makes sense to also log it in destructor (for example, it can be 1.5 GB). I also don't think we need this complication in the code.

Reply:
It's to avoid showing the following duplications in server log:

```
{b8c33030-a0d6-47db-b22c-27bf965ced36} <Debug> executeQuery: (from [::1]:35832) select * from numbers(10) (stage: Complete)
{b8c33030-a0d6-47db-b22c-27bf965ced36} <Trace> ContextAccess (default): Access granted: CREATE TEMPORARY TABLE ON *.*
{b8c33030-a0d6-47db-b22c-27bf965ced36} <Trace> InterpreterSelectQuery: FetchColumns -> Complete
{b8c33030-a0d6-47db-b22c-27bf965ced36} <Information> executeQuery: Read 10 rows, 80.00 B in 0.000598944 sec., 16696 rows/sec., 130.44 KiB/sec.
{b8c33030-a0d6-47db-b22c-27bf965ced36} <Debug> MemoryTracker: Peak memory usage (for query): 8.74 KiB.
{b8c33030-a0d6-47db-b22c-27bf965ced36} <Debug> TCPHandler: Processed in 0.000911912 sec.
{} <Debug> MemoryTracker: Peak memory usage (for query): 8.74 KiB.  <--- duplicate here
```

P.S. We already have such deduplication logic: https://github.com/ClickHouse/ClickHouse/blob/295f0a0557d3771384297298a0b8063f87a6991c/src/Interpreters/ThreadStatusExt.cpp#L623-L624